### DISCUSSION TITLE
Custom command after build

### DISCUSSION THREAD

**Discussion by MartinOrell** [secure-dev-ops/code-realtime#18] [Ideas]
Is there a way to make custom commands after a build?
What I'm trying to achieve is adding a command to copy folders and/or files into the same folder as the executable after a build. 

For reference in a C++ projects with cmake I can copy my _res_ folder with this in the Cmakelists file:
```
add_custom_command(TARGET main POST_BUILD
    COMMAND ${CMAKE_COMMAND} -E copy_directory
        ${CMAKE_SOURCE_DIR}/res $<TARGET_FILE_DIR:${PROJECT_NAME}>/res
)
```
But I don't see a way to do this with the Transformation Configuration file used in Code RealTime.

**Comment by mattias-mohlin** [comment]
It's not yet supported to customize the generated makefile like this, but it's something we want to support in the future. In Model RealTime we support a special TC property "compilationMakeInsert" but it's not yet supported in Code RealTime, and also I'm not sure it would have been enough for realizing post-build actions. We'll add this idea to our roadmap.

**Reply by MartinOrell** [reply]
ok, sounds good. In my use case it would not be a big deal whether it would be pre-build or post-build. I guess both pre-build and post-build actions would be nice features to have though.

**Reply by mattias-mohlin** [reply]
There might be one thing you can try: set the "makeCommand" property to invoke a script. Then inside that script you can perform the make command followed by any other actions you want to perform.
Here is an example of how to change the make command to just print a message when you do the build:
```js
tc.makeCommand = 'echo "Building..."';
```

**Reply by MartinOrell** [reply]
That actually works!
I did it with:
```
tc.makeCommand = 'make; cp -R ../../HelloWorld/res res'
```
